frontend-react/src/components/ManagePublicKey/ManagePublicKey.tsx

```diff
@@ -1,18 +1,148 @@
-import React from "react";
-import { GridContainer } from "@trussworks/react-uswds";
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { GridContainer, Icon, SiteAlert } from "@trussworks/react-uswds";
 
 import { AuthElement } from "../AuthElement";
 import { withCatchAndSuspense } from "../RSErrorBoundary";
+import { USLink } from "../USLink";
+import Spinner from "../Spinner";
+import { showError } from "../AlertNotifications";
 import { MemberType } from "../../hooks/UseOktaMemberships";
+import {
+    validateFileType,
+    ManagePublicKeyAction,
+    CONTENT_TYPE,
+    FORMAT,
+    validateFileSize,
+} from "../../hooks/network/ManagePublicKey/ManagePublicKeyHooks";
+
+import { ManagePublicKeyChooseSender } from "./ManagePublicKeyChooseSender";
+import { ManagePublicKeyUpload } from "./ManagePublicKeyUpload";
+
+const LightbulbIcon = Icon.Lightbulb;
+
+const ManagePublicKeySwitchDisplay = () => {
+    const navigate = useNavigate();
+    const [action, setAction] = useState(ManagePublicKeyAction.SELECT_SENDER);
+    const [fileContent, setFileContent] = useState("");
+    const [fileName, setFileName] = useState("");
+
+    // TODO: mocked for now - make the call you need when sending the file
+    const { sendFile } = {
+        sendFile: (data: {
+            contentType?: string;
+            fileContent?: string;
+            fileName?: string;
+        }) => {
+            data = { fileName: "filename" };
+            return data;
+        },
+    };
+
+    const onSenderSelect = (selectedSender: string) => {
+        if (selectedSender !== "") {
+            setAction(ManagePublicKeyAction.SELECT_PUBLIC_KEY);
+        }
+    };
+
+    const onBack = () => {
+        navigate("/resources");
+    };
+
+    const onPublicKeySubmit = async (
+        event: React.FormEvent<HTMLFormElement>
+    ) => {
+        event.preventDefault();
+
+        if (fileContent.length === 0) {
+            showError("No file contents to validate.");
+            return;
+        }
+
+        try {
+            setAction(ManagePublicKeyAction.SAVE_PUBLIC_KEY);
+            sendFile({
+                contentType: CONTENT_TYPE,
+                fileContent: fileContent,
+                fileName: fileName,
+            });
+            // based on !isProcessing
+            setAction(ManagePublicKeyAction.PUBLIC_KEY_SAVED);
+        } catch (e: any) {
+            console.trace(e);
+            showError(`Uploading public key failed. ${e.toString()}`);
+            setAction(ManagePublicKeyAction.SELECT_PUBLIC_KEY);
+        }
+    };
+
+    const onFileChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+        // No file selected
+        if (!event?.target?.files?.length) return;
```

Comment:
Do we need to check against `files.length` and `files.item` for file falsiness?

Reply:
No, but if we don't have this `if (!file) return; ` check then typescript complains that file maybe null.